lxd/instance/drivers/driver_qemu.go

```diff
@@ -6492,7 +6492,17 @@ func (d *qemu) checkFeatures(qemu string) ([]string, error) {
 
 func (d *qemu) Metrics(hostInterfaces []net.Interface) (*metrics.MetricSet, error) {
 	if d.agentMetricsEnabled() {
-		return d.getAgentMetrics()
+		metrics, err := d.getAgentMetrics()
+		if err != nil {
+			if !errors.Is(err, errQemuAgentOffline) {
+				d.logger.Warn("Could not get VM metrics from agent", logger.Ctx{"err": err})
```

Comment:
Naive question: will this be logged on every metric poll triggering the fallback?

Reply:
Yes, but not if the agent isn't registered as running (thus it would be a `errQemuAgentOffline` error) so there would have to be something unexpectedly failing. This change actually reduces the log entries as it is logging a warning right now on every request if the lxd-agent isn't running (which is what lead me to this change and thus the fallback scenario).